cogs/games.py

```diff
@@ -0,0 +1,106 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) idoneam (2016-2019)
+#
+# This file is part of Canary
+#
+# Canary is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Canary is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Canary. If not, see <https://www.gnu.org/licenses/>.
+
+# discord-py requirements
+import discord
+from discord.ext import commands
+
+# Other utilities
+import random
+import re
+from .utils.dnd_roll import dnd_roll
+
+
+class Games(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+
+    @commands.command()
+    async def roll(self, ctx, arg: str = '', mpr: str = ''):
+        """
+        Performs a DnD-style diceroll
+        Supports modifiers, multiple rolls, any-sided dice
+        """
+        roll_pattern = re.compile(r'^(\d*)d(\d*)([+-]?\d*)$')
+        roll_cmd = roll_pattern.match(arg)
+        params = {
+            'sides': 20,    # Default parameters
+            'repeat': 1,
+            'mod': 0
+        }
+        if arg == 'safe':    # nice meme bro
+            await ctx.send('https://i.imgur.com/2icUGpc.png')
+            return
+        if arg == 'help' or arg == 'h':
+            helpmsg = discord.Embed(
```

Comment:
This is a constant embed (i.e. it never changes) so it could be initialized as `HELP_MSG` outside the cog or outside the method to not re-define it every time and to keep the method itself small

Reply:
Resolved in b6899e4